annotation-processor/src/main/java/mekanism/ParamNameMapper.java

```diff
@@ -96,14 +81,11 @@ public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment
                         // annotations that this element possesses
                         for (Class<? extends Annotation> annotation : scanner.supportedAnnotations()) {
                             if (annotatedElement.getAnnotation(annotation) != null) {
-                                JsonObject classMethods = annotatedData.computeIfAbsent(scanner, a -> new JsonObject());
-                                JsonObject methods = getOrAddObject(classMethods, className);
-                                JsonObject signatures = getOrAddObject(methods, methodName);
-                                if (paramNames.size() == 1) {//Flatten the array to a single element
-                                    signatures.addProperty(methodSignature, paramNames.get(0).getAsString());
-                                } else {
-                                    signatures.add(methodSignature, paramNames);
-                                }
+                                scanner.originatingElements().add(annotatedElement);
+                                var classMethods = annotatedData.computeIfAbsent(scanner, a -> new TreeMap<>());
+                                var methods = classMethods.computeIfAbsent(className, a->new TreeMap<>());
+                                var signatures = methods.computeIfAbsent(methodName, a->new TreeMap<>());
```

Comment:
this logic is the same as you wrote, all I did was remove the Gson stuff as it was a bit harder to understand the structure. It could be changed to some kind of `anyMatch` logic on the supported annotations.

inlined some of the `computeIfAbsent` though

Reply:
The change you made looks decent, though do we want to move the declaration/creation of the `signatures` map outside of the `scanner.supportedAnnotations()` loop? (potentially making it lazy if adding empty elements to that map may make things behave wrongly)

Reply:
lets deal with that outside this PR - it was how your logic did things and this change to regular maps probably should have been done separately
I believe the map would have been done like this so that only the relevant param names ended up in the relevant file